src/lib/hooks/routing/useClientSideSmartOrderRouterTrade.ts

```diff
@@ -80,11 +81,13 @@ export default function useClientSideSmartOrderRouterTrade<TTradeType extends Tr
     }
   }, [config, params, queryArgs, wrapType])
 
-  const { data, error } = usePoll(getQuoteResult, JSON.stringify(queryArgs), isDebouncing) ?? {
+  const getIsFreshBlock = useGetIsFreshBlock()
+  const { data: quoteResult, error } = usePoll(getQuoteResult, JSON.stringify(queryArgs), {
+    debounce: isDebouncing,
+    staleCallback: useCallback(({ data }) => getIsFreshBlock(Number(data.blockNumber) || 0), [getIsFreshBlock]),
```

Comment:
will `data` always be defined here?

Reply:
No! You're right. This should read `data?.blockNumber`. `useCallback` seems to erase the type.